File: Pythonleetcode/Leetcode1678.py
class Solution(object):
    def interpret(self, command):
        str_index=0
        ans=""
        while str_index<len(command):
            if command[str_index]=='(':
                if command[str_index+1]=="a":
                    ans+="al"
                    str_index+=4
                else:
                    ans += "o"
                    str_index += 2
            elif command[str_index]=='G':
                ans += "G"
                str_index += 1

        return ans

print(Solution().interpret(command = "G()(al)"))
print(Solution().interpret(command = "G()()()()(al)"))
print(Solution().interpret(command = "(al)G(al)()()G"))

File: CppPractice/Project1/Project1/Leetcode1678.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

using namespace std;

class Solution {
public:
    string interpret(string command) {

        int index = 0;
        string ans;
        while (index < command.size())
        {
            if (command[index] == 'G')
            {
                ans += "G";
                index += 1;

            }
            else if (command[index] == '(' and command[index + 1] == ')')
            {
                ans += "o";
                index += 2;

            }
            else  if (command[index] == '(' and command[index + 1] == 'a')
            {
                ans += "al";
                index += 4;
            }

        }

        return ans;


    }
};